expression/builtin_miscellaneous.go

```diff
@@ -505,9 +506,16 @@ func (b *builtinInetAtonSig) evalInt(row chunk.Row) (int64, bool, error) {
 	if err != nil || isNull {
 		return 0, true, err
 	}
+	errWrongValueForType := errWrongValueForType.GenWithStackByArgs("string", val, ast.InetAton)
+	stmtCtx := b.ctx.GetSessionVars().StmtCtx
+	// Only in select, we will show warnings rather than error
+	truncateAsWarning := !(stmtCtx.InInsertStmt || stmtCtx.InUpdateStmt || stmtCtx.InDeleteStmt)
```

Comment:
I belive `TruncateAsWarning` is a parameter for client, e.g. `sc.TruncateAsWarning = (flags & model.FlagTruncateAsWarning) > 0`. It should be set here.